### PR TITLE
Refactor and simplify mocks

### PR DESCRIPTION
**Pull request by tsundvoll**

### CODE REVIEW

File: src/isar/config/predefined_poses/predefined_poses.py [hunk 1]
```diff
@@ -303,7 +297,7 @@
         frame=Frame.Asset,
     ),
     # EJE-342-1004.03 Robot orientation 90 degrees
-    "EJE-342-0226.10": Pose(
+    "EJE-342-1004.03": Pose(
```
Comment (oysand) [inline src/isar/config/predefined_poses/predefined_poses.py:300]: Why is this tag changed?

File: src/isar/config/predefined_poses/predefined_poses.py [hunk 2]
```diff
@@ -315,7 +309,7 @@
         frame=Frame.Asset,
     ),
     # JBZS-313-021 Robot orientation 270 degrees
-    "JBZS-313-024": Pose(
+    "JBZS-313-021": Pose(
```
Comment (oysand) [inline src/isar/config/predefined_poses/predefined_poses.py:312]: Why is this tag changed?

File: tests/mocks/pose.py [hunk 3]
```diff
@@ -0,0 +1,14 @@
...
+from robot_interface.models.geometry.pose import Pose
+from robot_interface.models.geometry.position import Position
+
+
+class MockPose:
+    default_pose = Pose(
+        position=Position(x=1, y=1, z=1, frame=Frame.Robot),
+        orientation=Orientation(  # Rotated 90 degrees about z-axis
+            x=0, y=0, z=0.7071068, w=0.7071068, frame=Frame.Robot
+        ),
```
Comment (Christdej) [inline tests/mocks/pose.py:12]:
```suggestion
        position=Position(x=0, y=0, z=0, frame=Frame.Robot),
        orientation=Orientation( 
            x=0, y=0, z=0, w=1, frame=Frame.Robot
        ),
```

Maybe this is slightly easier?

File: tests/isar/services/readers/test_base_reader.py [hunk 4]
```diff
@@ -8,37 +8,19 @@
...
-        assert Utilities.compare_two_arguments(content, expected_output)
-
     @pytest.mark.parametrize(
         "dataclass_dict, expected_dataclass",
         [
-            (asdict(mock_mission_definition("long_mission")), Mission),
-            (asdict(MockTask.drive_to()), Task),
-            (asdict(MockTask.take_image_in_coordinate_direction()), Task),
-            (asdict(mock_pose()), Pose),
+            (asdict(MockMissionDefinition.default_mission), Mission),
```
Comment (oysand) [inline tests/isar/services/readers/test_base_reader.py:20]: Does this stil use the "long mission"? Not sure if it needs to

File: tests/mocks/task.py [hunk 5]
```diff
@@ -6,19 +6,16 @@
...
-    ) -> DriveToPose:
-        return DriveToPose(
-            pose=Pose(position=position, orientation=orientation, frame=Frame.Robot)
+    drive_to = DriveToPose(
+        pose=Pose(
+            position=Position(x=1, y=1, z=1, frame=Frame.Robot),
+            orientation=Orientation(
+                x=0, y=0, z=0.7071068, w=0.7071068, frame=Frame.Robot
+            ),
+            frame=Frame.Robot,
```
Comment (Christdej) [inline tests/mocks/task.py:15]: Is it possible to use the MockPose you made in mock/pose.py?

**Review by Christdej — CHANGES_REQUESTED** [review]

File: src/isar/config/predefined_poses/predefined_poses.py (hunk 1 above)
Comment (tsundvoll) [reply src/isar/config/predefined_poses/predefined_poses.py:300]: The tag was changed to match the comment above. There was two entries with key "EJE-342-0226.10", so I assumed this one was a typo.

File: src/isar/config/predefined_poses/predefined_poses.py (hunk 2 above)
Comment (tsundvoll) [reply src/isar/config/predefined_poses/predefined_poses.py:312]: Same as above

File: tests/isar/services/readers/test_base_reader.py (hunk 4 above)
Comment (tsundvoll) [reply tests/isar/services/readers/test_base_reader.py:20]: I removed the long mission. It was nothing fundamentally different from the default mission. Correct me if the long mission is still needed.

File: src/isar/config/predefined_poses/predefined_poses.py (hunk 1 above)
Comment (oysand) [reply src/isar/config/predefined_poses/predefined_poses.py:300]: That seams reasonable

File: src/isar/config/predefined_poses/predefined_poses.py (hunk 2 above)
Comment (oysand) [reply src/isar/config/predefined_poses/predefined_poses.py:312]: That seams reasonable

File: tests/isar/services/readers/test_base_reader.py (hunk 4 above)
Comment (oysand) [reply tests/isar/services/readers/test_base_reader.py:20]: Agreed that it should not be needed

File: tests/mocks/pose.py (hunk 3 above)
Comment (tsundvoll) [reply tests/mocks/pose.py:12]: I agree. Currently, no reason not to use the simplest mock for pose.

File: tests/mocks/task.py (hunk 5 above)
Comment (tsundvoll) [reply tests/mocks/task.py:15]: Agreed, using the `MockPose.default_pose` as you suggest.

**Review by Christdej — APPROVED** [review]

**Pull request merged**